graylog2-server/src/main/java/org/graylog/events/notifications/EventNotificationExecutionJob.java

```diff
@@ -58,21 +60,21 @@
     private final DBEventDefinitionService eventDefinitionService;
     private final DBNotificationGracePeriodService notificationGracePeriodService;
     private final Map<String, EventNotification.Factory> eventNotificationFactories;
-    private final long retryTimeInMinutes;
-
+    private final EventsConfiguration eventsConfiguration;
 
     @Inject
     public EventNotificationExecutionJob(@Assisted JobDefinitionDto jobDefinition,
                                          DBNotificationService dbNotificationService,
                                          DBEventDefinitionService eventDefinitionService,
                                          DBNotificationGracePeriodService notificationGracePeriodService,
-                                         Map<String, EventNotification.Factory> eventNotificationFactories) {
+                                         Map<String, EventNotification.Factory> eventNotificationFactories,
+                                         EventsConfigurationProvider configurationProvider) {
         this.jobConfig = (Config) jobDefinition.config();
         this.notificationService = dbNotificationService;
         this.eventDefinitionService = eventDefinitionService;
         this.notificationGracePeriodService = notificationGracePeriodService;
         this.eventNotificationFactories = eventNotificationFactories;
-        this.retryTimeInMinutes = 1L; // TODO: Make retry time configurable in the plugin settings
+        this.eventsConfiguration = configurationProvider.get();
```

Comment:
Using `.get()` on the provider in the constructor can be problematic. If it throws an exception, users will get an ugly injection exception. If we call it in the execute method, we can catch the exception and log a nicer error.